### ISSUE TITLE
--compiler option appends ++ to every 'clang' match in the path.

### ISSUE THREAD

**Issue by phcerdan** [rdnetto/YCM-Generator#27]
My clang executable is ~/devtoolset/clang/bin/clang++
executing in build folder (cmake), with clang already in $PATH
~/.vim/bundle/YCM-generator/config_gen.py . --verbose -e --compiler ~/devtoolset/clang/bin/clang++
which: no clang++++ in (~/devtoolset/clang++/bin)

Or:
~/.vim/bundle/YCM-generator/config_gen.py . --verbose -e --compiler ~/devtoolset/clang/bin
which: no clang++ in (~/devtoolset/**clang++**/bin)

It is appending ++ in every match, not only at the final clang.
Not much time to solve it, sorry! Hope it helps

**Comment by ssfdust** [comment]
here it behaves normally.
I think you need to debug it with some debugger, like pudb.

```
ycm_generator . --verbose -e --compiler ~/devtoolset/clang/bin/clang++                         
which: no clang++ in (/home/ssfdust/devtoolset/clang/bin)
ERROR: Could not find clang at '/home/ssfdust/devtoolset/clang/bin/clang++'. Please make sure it is installed and is either in your path, or specified with --compiler.
```

**Comment by rdnetto** [comment]
This has been fixed on the development branch - see #21. Try that branch and if it still doesn't work, let me know.

Sorry for the delay in replying, I just moved interstate and have had my plate full until now.
